### DISCUSSION TITLE
Need help with error when trying to run multitask

### DISCUSSION THREAD

**Discussion by naruto1980** [pybricks/support#1935] [Q&A] [answered]
I am new to multitask so need your help
I try to make the robot move why the same time moving the attachment with code below

```
# Define parallel task action
async def Flip_Motion():
    await multitask(drive_base.straight(-50),front_motor.run_angle(100,100))
```

and call this function using
`run_task(Flip_Motion)`

I would like to understand if I miss anything but I got the error below:
```
Traceback (most recent call last):
  File "Cardinal_Path_4.py", line 97, in <module>
TypeError: 'generator' object isn't iterable
```

**Comment by BertLindeman** [comment]
Hello,

Looking into the Pybricks [documentation for multitask](https://code.pybricks.com/static/docs/v2.19.0/tools/index.html#pybricks.tools.multitask)
 I see:
```await multitask(coroutine1, coroutine2, ...)→ Tuple```

In your multitask command you have a number of commands, that should be  coroutines, so defined functions to solve this syntax error.

So it might be something like:
```
await def robotmove():
   # await do something

await def attachmentmove():
   # await do something else

```

**Comment by laurensvalk** [comment, marked as answer]
The parenthesis are missing:
```python
run_task(Flip_Motion)
```
It should be:
```python
run_task(Flip_Motion())
```

**Reply by naruto1980** [reply]
This solved my problem. Thank you.
I try to have a command with wait = False for example
drive_base.straight(100, wait=False) 
this command will not work.
